### PR TITLE
incusd/device/tpm: Wait for swtpm to be ready

### PR DESCRIPTION
**Pull request by stgraber**

### CODE REVIEW

File: internal/server/device/tpm.go [hunk 1]
```diff
@@ -227,6 +227,9 @@ func (d *tpm) startVM() (*deviceConfig.RunConfig, error) {
 		return nil, fmt.Errorf("Failed to save swtpm state for device %q: %w", d.name, err)
 	}
 
+	// swtpm unfortunately isn't always immediately ready for QEMU...
+	time.Sleep(time.Second)
```
Comment (breml) [inline internal/server/device/tpm.go:231]: I wonder, if there isn't a nicer way to make sure, that the swtpm is indeed ready. With this sleep, we always loose 1 second here. On the other hand, waiting 1 second might also not always be enough for the swtmp to become ready e.g. on a heavily loaded or very minimal system.

Comment (stgraber) [reply internal/server/device/tpm.go]: I was somehow remembering our older implementation where we were passing an fd which would have been pretty problematic to monitor, but given that following a fix for a path length issue, we're now using a regular socket, we can just wait for it to exist (after ensuring it doesn't exist prior to starting up).

File: internal/server/device/tpm.go [hunk 2]
```diff
@@ -227,6 +230,21 @@ func (d *tpm) startVM() (*deviceConfig.RunConfig, error) {
 		return nil, fmt.Errorf("Failed to save swtpm state for device %q: %w", d.name, err)
 	}
 
+	// Wait for the socket to be available.
+	exists := false
+	for i := 0; i < 20; i++ {
+		if util.PathExists(socketPath) {
+			exists = true
+			continue
```
Comment (gibmat) [inline internal/server/device/tpm.go:238]: Wouldn't `break` be better?

Comment (stgraber) [reply internal/server/device/tpm.go:238]: doh, yes, that's what I meant to put there ;)

Comment (stgraber) [reply internal/server/device/tpm.go]: Fixed

**Pull request merged**